src/FileFormats.Minidump/MinidumpVirtualAddressSpace.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FileFormats.Minidump
+{
+    public class MinidumpVirtualAddressSpace : IAddressSpace
+    {
+        private IAddressSpace _addressSpace;
+        private ReadOnlyCollection<MinidumpSegment> _segments;
+        private ulong _length;
+
+        public ulong Length
+        {
+            get
+            {
+                return _length;
+            }
+        }
+
+        public MinidumpVirtualAddressSpace(ReadOnlyCollection<MinidumpSegment> segments, IAddressSpace addressSpace)
+        {
+            _addressSpace = addressSpace;
+            _segments = segments;
+            MinidumpSegment last = segments.Last();
+            _length = last.VirtualAddress + last.Size;
+        }
+
+        public uint Read(ulong position, byte[] buffer, uint bufferOffset, uint count)
+        {
+            if (count == 0)
+                return 0;
+            
+            MinidumpSegment seg = FindSegment(position);
+            if (seg == null)
+                return 0;
+            
+            // TODO: What if they read past the end of the segment?
```

Comment:
The current ELF implementation was written before I had created the PiecewiseAddressSpace. I agree the linear search isn't ideal, but it could easily be replaced with a binary search as you have done here. I'm not opposed to checking this in as-is, but it seems like in the long run we don't need 3 or 4 different implementations of VirtualAddressing that only differ in terms of what file format data structure held the mapping info.